src/chaos.js

```diff
@@ -9,39 +9,78 @@ export class KubernetesClient {
         }
     }
 
-    getPodNames(nameSpace) {
-        return this.kubernetes.pods.list(nameSpace).map(function(pod){
+    getPodNames(namespace) {
+        return this.kubernetes.pods.list(namespace).map(function(pod){
             return pod.name
         })
     }
 
-    getJobNames(nameSpace) {
-        return this.kubernetes.jobs.list(nameSpace).map(function(job){
+    getJobNames(namespace) {
+        return this.kubernetes.jobs.list(namespace).map(function(job){
             return job.name
         })
     }
 
-    getRandomPod(nameSpace) {
-        const podNames = this.getPodNames(nameSpace)
+    getNodeNames() {
+        return this.kubernetes.nodes.list().map(function(node){
+            return node.name
+        })
+    }
+
+    getRandomPod(namespace) {
+        const podNames = this.getPodNames(namespace)
         return podNames[Math.floor(Math.random() * podNames.length)]
     }
 
-    getRandomJob(nameSpace) {
-        const jobNames = this.getJobNames(nameSpace)
+    getRandomJob(namespace) {
+        const jobNames = this.getJobNames(namespace)
         return jobNames[Math.floor(Math.random() * jobNames.length)]
     }
 
+    runJobOnNodes(namespace, jobName, nodes, image, command){
+        for (let i = 0; i < nodes.length; i++) {
+          let name = `${jobName}-${(Math.random() + 1).toString(36).substring(7)}`;
+          console.log(`Creating job ${name} on node ${nodes[i]}`)
+          this.kubernetes.jobs.create({
+            namespace: namespace,
+            name: name,
+            image: image,
+            command: command,
+            node_name: nodes[i]
+          })
+        }
+        return jobName
+    }
+
+    cleanJobs(namespace, jobName){
+        let jobs = this.getJobNames(namespace)
+        for (let i = 0; i < jobs.length; i++) {
+            if (jobs[i].includes(jobName)) {
+                console.log(`Deleting job ${jobs[i]}`)
+                this.kubernetes.jobs.delete(jobs[i],namespace)
+            }
+        }
+        let pods = this.getPodNames(namespace)
+        for (let i = 0; i < pods.length; i++) {
+            if (pods[i].includes(jobName)) {
+                console.log(`Deleting pod ${pods[i]}`)
+                this.kubernetes.pods.delete(pods[i],namespace)
+            }
+        }
+        return jobName
+    }
+
     // Kill a random pod in the specified namespace.
-    killRandomPod(nameSpace) {
-        const podName = this.getRandomPod(nameSpace)
-        this.kubernetes.pods.kill(podName,nameSpace)
+    killRandomPod(namespace) {
+        const podName = this.getRandomPod(namespace)
+        this.kubernetes.pods.kill(podName,namespace)
         return podName
     }
 
     // Kill a random job in the specified namespace.
-    killRandomJob(nameSpace) {
-        const jobName = this.getRandomJob(nameSpace)
-        this.kubernetes.jobs.kill(jobName,nameSpace)
+    killRandomJob(namespace) {
```

Comment:
What is the purpose of this function? Contrary to `killRandomPod` I cannot come to a use case for it.